internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,185 @@ package github_connection
+	}
+	pullRequestNumber, err := strconv.Atoi(pullRequestNumberStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid GITHUB_PR_NUMBER: %w", err)
+	}
 
-	GithubClient := github.NewClient(nil).WithAuthToken(params.Token)
+	eventName := os.Getenv("GITHUB_EVENT_NAME")
+	if eventName == "" {
+		return nil, fmt.Errorf("GITHUB_EVENT_NAME environment variable is not set")
+	}
+	GithubClient := github.NewClient(nil).WithAuthToken(token)
 
 	return &GithubConnection{
-		client:            GithubClient,
-		RepositoryName:    params.RepositoryName,
-		RepoOwner:         params.RepoOwner,
-		PullRequestNumber: params.PullRequestNumber,
+		Client:            GithubClient,
+		RepositoryName:    githubRepositoryName,
+		RepoOwner:         repoOwner,
+		PullRequestNumber: pullRequestNumber,
+		EventName:         eventName,
 	}, nil
 }
 
-type GithubConnection struct {
-	client            *github.Client
-	RepositoryName    string
-	RepoOwner         string
-	PullRequestNumber int
+func (receiver *GithubConnection) GetEventName() string {
+	return receiver.EventName
 }
 
-func (receiver *GithubConnection) GetRepositoryName() string {
+func (receiver *GithubConnection) GetRepository() string {
 	return receiver.RepositoryName
 }
 
-func (receiver *GithubConnection) GetRepository() string {
-	return receiver.RepositoryName
+func (receiver *GithubConnection) GetPullRequestChanges() ([]github.CommitFile, error) {
+	var ctx = context.Background()
+	var opt = &github.ListOptions{PerPage: 100}
+	var allChanges []github.CommitFile
+
+	for {
+		files, resp, err := receiver.Client.PullRequests.ListFiles(
+			ctx, receiver.RepoOwner,
+			receiver.RepositoryName,
+			receiver.PullRequestNumber,
+			opt)
+
+		if err != nil {
+			return nil, fmt.Errorf("error listing pull request files: %w", err)
+		}
+		for _, file := range files {
+			if file.GetAdditions() == 0 {
+				continue
+			}
+			allChanges = append(allChanges, *file)
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+
+	return allChanges, nil
 }
 
-func (receiver GithubConnection) CreateComment(comment string) error {
+func (receiver *GithubConnection) CreateComment(files []github.CommitFile) error {
 	ctx := context.Background()
 
-	comment_text := &github.IssueComment{
-		Body: github.String(comment),
+	for _, file := range files {
+		fmt.Printf("Filename: %s\n", file.GetFilename())
+		fmt.Printf("Patch: %s\n", file.GetPatch())
+
+		comments, err := analyzeFileAndCreateComments(&file)
+		if err != nil {
+			return fmt.Errorf("error analyzing file %s: %w", file.GetFilename(), err)
+		}
+
+		for _, commentData := range comments {
+			comment, _, err := receiver.Client.PullRequests.CreateComment(
+				ctx,
+				receiver.RepoOwner,
+				receiver.RepositoryName,
+				receiver.PullRequestNumber,
+				commentData)
+			if err != nil {
+				return fmt.Errorf("error creating PullRequest comment: %w", err)
+			}
+			fmt.Printf("Review comment created: %s\n", comment.GetHTMLURL())
+		}
 	}
```

Comment:
Code Review 2024-08-18 02:28:52 - New lines 86 to 106:

	for _, file := range files {
		fmt.Printf("Filename: %s\n", file.GetFilename())
		fmt.Printf("Patch: %s\n", file.GetPatch())

		comments, err := analyzeFileAndCreateComments(&file)
		if err != nil {
			return fmt.Errorf("error analyzing file %s: %w", file.GetFilename(), err)
		}

		for _, commentData := range comments {
			comment, _, err := receiver.Client.PullRequests.CreateComment(
				ctx,
				receiver.RepoOwner,
				receiver.RepositoryName,
				receiver.PullRequestNumber,
				commentData)
			if err != nil {
				return fmt.Errorf("error creating PullRequest comment: %w", err)
			}
			fmt.Printf("Review comment created: %s\n", comment.GetHTMLURL())
		}